packages/@sanity/portable-text-editor/src/utils/validateValue.ts

```diff
@@ -1,23 +1,17 @@
+import {PortableTextBlock, PortableTextSpan, PortableTextTextBlock} from '@sanity/types'
 import {flatten, isObject, uniq} from 'lodash'
 import {set, unset, insert} from '../patch/PatchEvent'
-import {PortableTextBlock, PortableTextChild, PortableTextFeatures} from '../types/portableText'
-import {InvalidValueResolution} from '../types/editor'
+import {InvalidValueResolution, PortableTextMemberTypes} from '../types/editor'
 
 export function validateValue(
   value: PortableTextBlock[] | undefined,
-  portableTextFeatures: PortableTextFeatures,
+  types: PortableTextMemberTypes,
   keyGenerator: () => string
 ): {valid: boolean; resolution: InvalidValueResolution | null} {
   let resolution: InvalidValueResolution | null = null
   let valid = true
-  const validChildTypes = [
-    ...[portableTextFeatures.types.span.name],
-    ...portableTextFeatures.types.inlineObjects.map((t) => t.name),
-  ]
-  const validBlockTypes = [
-    ...[portableTextFeatures.types.block.name],
-    ...portableTextFeatures.types.blockObjects.map((t) => t.name),
-  ]
+  const validChildTypes = [...[types.span.name], ...types.inlineObjects.map((t) => t.name)]
+  const validBlockTypes = [...[types.block.name], ...types.blockObjects.map((t) => t.name)]
 
```

Comment:
could be simplified to ```suggestion
const validBlockTypes = [types.block.name, ...types.blockObjects.map((t) => t.name)]
```